Libraries/LibLine/Editor.cpp

```diff
@@ -992,6 +993,16 @@ void Editor::refresh_display()
         m_cached_buffer_size = m_buffer.size();
         return;
     }
+    // We might be at the last line, and have more than one line;
+    // Refreshing the display will cause the terminal to scroll,
+    // so note that fact and bring origin up.
+    auto current_num_lines = num_lines();
+    if (m_origin_x + current_num_lines > m_num_lines + 1) {
```

Comment:
Adding together x (horizontal) and lines (vertical) doesn't look right. What's happening here?

Reply:
the x and y are inverted here, x is the vertical axis, y is the horizontal axis.
kinda odd, but :shrug:

Reply:
I'm not even sure why I named them like this, I could go through and fix their names.
will take but a moment.

Reply:
Yeah let's fix that. This is way confusing :)

Reply:
There we go.
no more confusion :)

Reply:
ugh there's more of them T-T